multidict/_multidict_base.py

```diff
@@ -34,34 +38,26 @@ def _abc_valuesview_register(view_cls: type[object]) -> None:
 
 
 def _viewbaseset_richcmp(
-    view: set[object], other: object, op: Literal[0, 1, 2, 3, 4, 5]
+    view: _ViewArg, other: object, op: Literal[0, 1, 2, 3, 4, 5]
 ) -> Union[bool, NotImplementedType]:
+    if not isinstance(other, Set):
+        return NotImplemented  # type: ignore[no-any-return]
```

Comment:
I don't understand what requires this type ignore. @Dreamsorcerer do you know? Or should the return type switch to something like
```suggestion
) -> Union[bool, type[ImplementedType]]:
    if not isinstance(other, Set):
        return NotImplemented
```
?

Reply:
Oh, so it must be `Any as NotImplementedType` that's at fault.

Reply:
Can we replace that import with `NotImplementedType = object`? Since we don't run MyPy against Python 3.9 anymore (which I still don't like, though), it wouldn't matter for the time being.

Reply:
Please hold on.
I feel this file will go away in a few days.

Reply:
It's the NotImplemented (not the type annotation). It seems to vary in some Python versions which is confusing, some Python versions it's reported as Any, and others it's not.

Regardless, that's not part of this PR, I already put that it. I don't think it's fixable at this point, but maybe it will change in a future Python/mypy release..